PolySTAR-Taproot-project/src/communication/cv_handler.hpp

```diff
@@ -0,0 +1,98 @@
+#ifndef CV_HANDLER_HPP_
+#define CV_HANDLER_HPP_
+
+#include <cstdint>
+#include <unordered_map>
+
+#include "tap/architecture/timeout.hpp"
+#include "tap/util_macros.hpp"
+
+#include "modm/container/deque.hpp"
+#include "modm/processing/protothread/semaphore.hpp"
+
+#include "cv_serial.hpp"
+#include "cv_serial_data.hpp"
+
+namespace src
+{
+class Drivers;
+}
+
+namespace src::communication::cv
+{
+/**
+ * A class designed to communicate with the 2023 version of the CV Jetson.
+ * Supports decoding various CV message types. Also supports sending
+ * feedback messages to the Jetson.
+ *
+ * @note use the instance stored in the `Drivers` to interact with this class
+ *      (you shouldn't be declaring your own `CVHandler` object).
+ *
+ * Receive information from the referee serial by continuously calling `messageReceiveCallback`.
+ * Access data sent by the referee serial by calling `getMovementData`, `getTurretData` or `getShootOrder`.
+ */
+class CVHandler : public CVSerial
+{
+public:
+    /**
+     * Constructs a CVHandler class connected to UART Port 7 with
+     * CRC enforcement disabled.
+     *
+     * @see `CVSerial` and `DJISerial`
+     */
+    CVHandler(Drivers* drivers);
+    DISALLOW_COPY_AND_ASSIGN(CVHandler)
+    mockable ~CVHandler() = default;
+
+    /**
+     * Handles the types of messages defined above in the RX message handlers section.
+     */
+    void messageReceiveCallback(const ReceivedSerialMessage& completeMessage) override;
+
+    /**
+     * Returns a reference to the most up to date turret setpoint struct.
+     */
+    mockable const Rx::TurretData& getTurretData() const { return turretData; };
+
+    /**
+     * Returns a reference to the most up to date movement setpoint struct.
+     */
+    mockable const Rx::MovementData& getMovementData() const { return movementData; };
+
+    /*
+    *  Returns value of flag indicating if CV has requested robot to fire
+    */
+    mockable bool getShootOrderFlag() const { return shootOrderFlag; };

```

Comment:
Why separate getting the shoot order and clearing it ? IMO from a logical standpoint the action of getting the flag should clear it to avoid forgetting to clear it after it has been received.